R/object_name_linters.R

```diff
@@ -1,3 +1,40 @@
+object_name_xpath <- paste0(
+  # assignments
+  "//SYMBOL[",
+  " not(preceding-sibling::OP-DOLLAR)",
+  " and ancestor::expr[",
+  "  following-sibling::LEFT_ASSIGN",
+  "  or preceding-sibling::RIGHT_ASSIGN",
+  "  or following-sibling::EQ_ASSIGN",
+  " ]",
+  " and not(ancestor::expr[",
+  "  preceding-sibling::OP-LEFT-BRACKET",
+  "  or preceding-sibling::LBB",
+  " ])",
+  "]",
+
+  " | ",
+
+  "//STR_CONST[",
+  " not(preceding-sibling::OP-DOLLAR)",
```

Comment:
`OP-AT` too for S4 extractions?

Reply:
Their assignment targets are `SLOT`, not `SYMBOL`, so we don't need to except these.